air/src/execution_step/instructions/ap.rs

```diff
@@ -46,7 +52,13 @@ impl<'i> super::ExecutableInstruction<'i> for Ap<'i> {
         let result = apply_to_arg(&self.argument, exec_ctx, trace_ctx, should_touch_trace)?;
 
         let merger_ap_result = to_merger_ap_result(self, trace_ctx)?;
-        let maybe_generation = populate_context(&self.result, &merger_ap_result, result, exec_ctx)?;
+        let maybe_generation = populate_context(
```

Comment:
I belive that this code could be simplified with introducing a new "internal" instruction for ap with three arguments

Reply:
Could you elaborate on the simplification?